### PR TITLE
Add aliases support

### PR DESCRIPTION
**Pull request by licht1stein**

### CODE REVIEW

File: obsidian.el [hunk 1]
```diff
@@ -147,19 +152,55 @@ If FILE is not specified, use the current buffer."
...
+	    (->> split
+		 (nth 1)
+		 yaml-parse-string)))))
+
+(defun obsidian-find-aliases-in-string (s)
+  "Return all aliases in S."
+  (let* ((front-matter (-> s
+			   obsidian-find-yaml-front-matter)))
+    (if front-matter
+	(gethash 'aliases front-matter))))
```
Comment (netromdk) [inline obsidian.el:186]: Can it do `'aliases` or `'alias` here?

File: obsidian.el [hunk 2]
```diff
@@ -393,6 +415,31 @@ See `markdown-follow-link-at-point' and
...
+			 cadr
+			 (plist-get :match)))
+	 (aliases  (->> aliases-s
+			(s-split ",")
+			(-map #'s-trim))))
+    (-map (lambda (alias) (list alias file)) aliases)))
+
+(defun obsidian--find-all-aliases ()
+  "Find all aliases in Obsidian directory."
+  (let* ((multiple (elgrep obsidian-directory "\.md" "^aliases: \\[\\(?1:.*\\)\\]" :recursive t :case-fold-search t :exclude-file-re "~" :abs t)))
```
Comment (netromdk) [inline obsidian.el:432]: Maybe do something similar to `\\(alias\\(es\\)?\\)` to match both "aliases" and "alias".

File: obsidian.el [hunk 3]
```diff
@@ -393,6 +415,31 @@ See `markdown-follow-link-at-point' and
...
+    (->> (-map #'obsidian--extract-alias multiple)
+	 -flatten
+	 (-partition 2))))
+
+(defun obsidian-update-aliases ()
+  "Update all aliasesx."
```
Comment (netromdk) [inline obsidian.el:438]: Stray "x" :)

**Comment by licht1stein** [comment]
@netromdk wow you're quick :)
test it now, should handle correctly both "alias" and "aliases" keys.

**Comment by netromdk** [comment]
@licht1stein this works great! The only thing I noticed is that it doesn't parse values like `000` correctly. It shows as `0`. But I admit it's a weird alias. ;-)

**Comment by licht1stein** [comment]
@netromdk put it in quotes, like `"000"`, shoud work then :)

**Comment by licht1stein** [comment]
Nope, still 0

**Comment by netromdk** [comment]
That doesn't work, unfortunately. It's not a deal breaker for me.

**Comment by netromdk** [comment]
🎉

**Comment by licht1stein** [comment]
Thanks for your help @netromdk 
By the way, I've copied a few things from your config, quite useful :)

**Comment by netromdk** [comment]
I'm glad to hear that :) Couldn't find yours(?). It's always fun to look at configs of others'.

**Comment by licht1stein** [comment]
I'm thinking about publishing it, but need to make sure there are no credentials inside :)

**Pull request merged**